frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx

```diff
@@ -311,46 +313,48 @@ function DashCard({
   ]);
 
   return (
-    <DashCardRoot
-      className="Card rounded flex flex-column hover-parent hover--visibility"
-      hasHiddenBackground={hasHiddenBackground}
-      isNightMode={isNightMode}
-      isUsuallySlow={isSlow === "usually-slow"}
-      ref={cardRootRef}
-    >
-      {renderDashCardActions()}
-      <DashCardVisualization
-        dashboard={dashboard}
-        dashcard={dashcard}
-        series={series}
-        parameterValues={parameterValues}
-        parameterValuesBySlug={parameterValuesBySlug}
-        metadata={metadata}
-        mode={mode}
-        gridSize={gridSize}
-        gridItemWidth={gridItemWidth}
-        totalNumGridCols={totalNumGridCols}
-        headerIcon={headerIcon}
-        expectedDuration={expectedDuration}
-        error={error}
-        isAction={isAction}
-        isEmbed={isEmbed}
-        isEditing={isEditing}
-        isEditingDashCardClickBehavior={isEditingDashCardClickBehavior}
-        isEditingDashboardLayout={isEditingDashboardLayout}
-        isEditingParameter={isEditingParameter}
-        isClickBehaviorSidebarOpen={isClickBehaviorSidebarOpen}
-        isSlow={isSlow}
-        isPreviewing={isPreviewingCard}
-        isFullscreen={isFullscreen}
+    <ErrorBoundary>
```

Comment:
That's a great improvement! I believe it would be even better to put it around in `Visualization` component to prevent page crashes in collections view, query builder, and publicly shared questions

Reply:
![why-not-both](https://user-images.githubusercontent.com/30528226/215515746-6243e4bb-d5d6-49a7-965e-e6330177e5e6.jpg)

I think there's some logic in dashcard that we want to catch errors on as well, but the beauty of error boundaries is we can wrap things at multiple levels.